flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/sink/KafkaSinkBuilder.java

```diff
@@ -0,0 +1,187 @@
+        // set the producer configuration properties for kafka record key value serializers.
+        if (!kafkaProducerConfig.containsKey(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG)) {
+            kafkaProducerConfig.put(
+                    ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                    ByteArraySerializer.class.getName());
+        } else {
+            LOG.warn(
+                    "Overwriting the '{}' is not recommended",
+                    ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG);
+        }
+
+        if (!kafkaProducerConfig.containsKey(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG)) {
+            kafkaProducerConfig.put(
+                    ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                    ByteArraySerializer.class.getName());
+        } else {
+            LOG.warn(
+                    "Overwriting the '{}' is not recommended",
+                    ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG);
+        }
+
+        if (!kafkaProducerConfig.containsKey(ProducerConfig.TRANSACTION_TIMEOUT_CONFIG)) {
+            final long timeout = DEFAULT_KAFKA_TRANSACTION_TIMEOUT.toMillis();
+            checkState(
+                    timeout < Integer.MAX_VALUE && timeout > 0,
+                    "timeout does not fit into 32 bit integer");
+            kafkaProducerConfig.put(ProducerConfig.TRANSACTION_TIMEOUT_CONFIG, (int) timeout);
+            LOG.warn(
+                    "Property [{}] not specified. Setting it to {}",
+                    ProducerConfig.TRANSACTION_TIMEOUT_CONFIG,
+                    DEFAULT_KAFKA_TRANSACTION_TIMEOUT);
+        }
+        return this;
+    }
+
+    /**
+     * Sets the {@link KafkaRecordSerializationSchema} that transforms incoming records to {@link
+     * org.apache.kafka.clients.producer.ProducerRecord}s.
+     *
+     * @param recordSerializer
+     * @return {@link KafkaSinkBuilder}
+     */
+    public KafkaSinkBuilder<IN> setRecordSerializer(
+            KafkaRecordSerializationSchema<IN> recordSerializer) {
+        this.recordSerializer = checkNotNull(recordSerializer, "recordSerializer");
+        ClosureCleaner.clean(
+                this.recordSerializer, ExecutionConfig.ClosureCleanerLevel.RECURSIVE, true);
+        return this;
+    }
+
+    /**
+     * Sets the prefix for all created transactionalIds if {@link DeliveryGuarantee#EXACTLY_ONCE} is
+     * configured.
+     *
+     * <p>It is mandatory to always set this value with {@link DeliveryGuarantee#EXACTLY_ONCE} to
+     * prevent corrupted transactions if multiple jobs using the KafkaSink run against the same
+     * Kafka Cluster. The default prefix is {@link #transactionalIdPrefix}.
+     *
+     * @param transactionalIdPrefix
+     * @return {@link KafkaSinkBuilder}
+     */
+    public KafkaSinkBuilder<IN> setTransactionalIdPrefix(String transactionalIdPrefix) {
+        this.transactionalIdPrefix = checkNotNull(transactionalIdPrefix, "transactionalIdPrefix");
```

Comment:
Please add some length check. I think the maximum kafka transaction id length is 64k (please double check), so we could cap at 60k and still have some leeway.